dbt/adapters/glue/gluedbapi/connection.py

```diff
@@ -276,7 +276,7 @@ def cursor(self, as_dict=False) -> GlueCursor:
         if self.state == GlueSessionState.READY:
             self._init_session()
             return GlueDictCursor(connection=self) if as_dict else GlueCursor(connection=self)
-        else:
```

Comment:
This change is not necessary. `not self.session_id or self.session_id is not None` always returns `True`.